`chezmoi archive` crashes for external of type `git-repo`

## Describe the bug

If .chezmoiexternal.yaml contains an external of type `git-repo` chezmoi archive command crashes

## To reproduce

1) Have an external of type `git-repo`  in .chezmoiexternal.yaml
2) Run chezmoi archive command

## Expected behavior

chezmoi archive should run successfully.

## Output of command with the `--verbose` flag

```console
$ chezmoi --verbose archive
panic: update to no update system

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/pkg/chezmoi.noUpdateSystemMixin.RunCmd(...)
	github.com/twpayne/chezmoi/v2/pkg/chezmoi/system.go:80
github.com/twpayne/chezmoi/v2/pkg/chezmoi.(*TargetStateModifyDirWithCmd).Apply(0xc000306108, {0x7f81dc37b4c0, 0xc000730c40}, {0x55b309be8248, 0xc000120270}, {0x55b309be4848?, 0xc0002c0170})
	github.com/twpayne/chezmoi/v2/pkg/chezmoi/targetstateentry.go:81 +0x126
github.com/twpayne/chezmoi/v2/pkg/chezmoi.(*SourceState).Apply(0xc0000ac000, {0x7f81dc37b4c0, 0xc000730c40}, {0x55b309bea4d0, 0xc000734cc0}, {0x55b309be8248, 0xc000120270}, {{0x0?, 0x55b3089d96f9?}}, {{0xc0006aa3f0, ...}}, ...)
	github.com/twpayne/chezmoi/v2/pkg/chezmoi/sourcestate.go:696 +0xbb2
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).applyArgs(0xc000096a80, {0x55b309be4298, 0xc00012c000}, {0x7f81dc37b4c0, 0xc000730c40}, {{0x0?, 0xedab58820?}}, {0xc000044cb0, 0x0, 0x1}, ...)
	github.com/twpayne/chezmoi/v2/pkg/cmd/config.go:622 +0x846
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).runArchiveCmd(0xc000096a80, 0xc00038a280, {0xc000044cb0, 0x0, 0x1})
	github.com/twpayne/chezmoi/v2/pkg/cmd/archivecmd.go:77 +0x6f0
github.com/spf13/cobra.(*Command).execute(0xc00038a280, {0xc000044ca0, 0x1, 0x1})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000310f00)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).execute(0xc00071fdf8?, {0xc00011c070, 0x2, 0x2})
	github.com/twpayne/chezmoi/v2/pkg/cmd/config.go:1189 +0x78
github.com/twpayne/chezmoi/v2/pkg/cmd.runMain({{0x55b309954c5c, 0x6}, {0x0, 0x0}, {0x55b309955660, 0x14}, {0x0, 0x0}}, {0xc00011c070, 0x2, ...})
	github.com/twpayne/chezmoi/v2/pkg/cmd/cmd.go:264 +0x1b4
github.com/twpayne/chezmoi/v2/pkg/cmd.Main({{0x55b309954c5c, 0x6}, {0x0, 0x0}, {0x55b309955660, 0x14}, {0x0, 0x0}}, {0xc00011c070, 0x2, ...})
	github.com/twpayne/chezmoi/v2/pkg/cmd/cmd.go:128 +0xb1
main.main()
	github.com/twpayne/chezmoi/v2/main.go:26 +0x10c
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
RESULT    CHECK                MESSAGE
warning   version              v2.22.1, built at 2022-09-05T03:33:46Z
ok        latest-version       v2.22.1
ok        os-arch              linux/amd64 (Arch Linux)
ok        uname                Linux sm1999-arch 5.15.67-1-lts #1 SMP Thu, 08 Sep 2022 12:59:52 +0000 x86_64 GNU/Linux
ok        go-version           go1.19 (gc)
ok        executable           /usr/bin/chezmoi
ok        config-file          no config file found
warning   source-dir           ~/.local/share/chezmoi is a git working tree (dirty)
ok        suspicious-entries   no suspicious entries
warning   working-tree         ~/.local/share/chezmoi is a git working tree (dirty)
ok        dest-dir             ~ is a directory
ok        shell-command        found /usr/bin/zsh
ok        shell-args           /usr/bin/zsh
ok        cd-command           found /usr/bin/zsh
ok        cd-args              /usr/bin/zsh
ok        edit-command         found /usr/bin/vi
ok        edit-args            /usr/bin/vi
info      diff-command         not set
ok        umask                022
ok        git-command          found /usr/bin/git, version 2.37.3
ok        merge-command        found /usr/bin/vimdiff
ok        age-command          found /usr/bin/age, version 1.0.0
ok        gpg-command          found /usr/bin/gpg, version 2.2.39
info      pinentry-command     not set
info      1password-command    op not found in $PATH
info      bitwarden-command    bw not found in $PATH
info      gopass-command       gopass not found in $PATH
ok        keepassxc-command    found /usr/bin/keepassxc-cli, version 2.7.1
info      keeper-command       keeper not found in $PATH
info      keepassxc-db         not set
info      lastpass-command     lpass not found in $PATH
info      pass-command         pass not found in $PATH
info      vault-command        vault not found in $PATH
info      secret-command       not set
```
</details>

--- a/pkg/chezmoi/tarwritersystem.go
+++ b/pkg/chezmoi/tarwritersystem.go
@@ -4,6 +4,7 @@ import (
 	"archive/tar"
 	"io"
 	"io/fs"
+	"os/exec"
 )
 
 // A TarWriterSystem is a System that writes to a tar archive.
@@ -34,6 +35,11 @@ func (s *TarWriterSystem) Mkdir(name AbsPath, perm fs.FileMode) error {
 	header.Name = name.String() + "/"
 	header.Mode = int64(perm)
 	return s.tarWriter.WriteHeader(&header)
+}
+
+// RunCmd implements System.RunCmd.
+func (s *TarWriterSystem) RunCmd(cmd *exec.Cmd) error {
+	return nil
 }
 
 // RunScript implements System.RunScript.

--- a/pkg/chezmoi/zipwritersystem.go
+++ b/pkg/chezmoi/zipwritersystem.go
@@ -4,6 +4,7 @@ import (
 	"archive/zip"
 	"io"
 	"io/fs"
+	"os/exec"
 	"time"
 )
 
@@ -38,6 +39,11 @@ func (s *ZIPWriterSystem) Mkdir(name AbsPath, perm fs.FileMode) error {
 	fileHeader.SetMode(fs.ModeDir | perm)
 	_, err := s.zipWriter.CreateHeader(&fileHeader)
 	return err
+}
+
+// RunCmd implements System.RunCmd.
+func (s *ZIPWriterSystem) RunCmd(cmd *exec.Cmd) error {
+	return nil
 }
 
 // RunScript implements System.RunScript.

--- a/pkg/cmd/testdata/scripts/externalgitrepo.txtar
+++ b/pkg/cmd/testdata/scripts/externalgitrepo.txtar
@@ -19,6 +19,12 @@ cd $WORK
 exec chezmoi apply
 cmp $HOME/.dir/.file golden/.file
 
+# test that chezmoi archive --format=tar creates an archive
+exec chezmoi archive --format=tar
+
+# test that chezmoi archive --format=zip creates an archive
+exec chezmoi archive --format=zip
+
 chhome home2/user
 mkgitconfig
 expandenv $WORK/home2/user/.local/share/chezmoi/.chezmoiexternal.toml

